numba/types/containers.py

```diff
@@ -359,6 +359,13 @@ def iterator_type(self):
     def is_precise(self):
         return self.dtype.is_precise()
 
+    def __getitem__(self, args):
+        """
+        Overrides the default __getitem__ from Type.
+        """
+        return self.dtype(dtype=self.dtype.dtype, ndim=self.dtype.ndim,
```

Comment:
Is it not just `self.dtype` that needs to be returned?

Reply:
probably, I was going to play around with the type system while setting up the tests.

Reply:
I would tend to agree with @stuartarchibald :)